invenio_base/utils.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+from flask import current_app
+from .signals import secret_key_changed
+from invenio_db import db
+from sqlalchemy import update
+from sqlalchemy_utils.types.encrypted import EncryptedType
+from sqlalchemy.engine import reflection
+
+
+def rebuild_encrypted_field(old_key, table, field):
+    """Rebuild an EncryptedType field when the SECRET_KEY is changed.
+
+    :param sender: signal sender object.
+    :param old_key: the old SECRET_KEY.
+    :param field: the field to change.
+    """
+    db.session.close_all()
+    eng = db.get_engine(current_app)
+    conn = eng.connect()
+    inspector = reflection.Inspector.from_engine(db.engine)
+    primary_key = inspector.get_primary_keys(table.__tablename__)[0]
+    query_result = conn.execute("""select %s, %s from %s;""" %
+                                (primary_key, field, table.__tablename__))
+    old_entries = query_result.fetchall()
+    decrypter = EncryptedType(type_in=db.Text, key=old_key)
```

Comment:
We should use directly the type specified in the model and just replace the `key`.